common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -662,6 +688,32 @@ def get_problem_html(self, encapsulate=True):
 
         return html
 
+    def remove_extra_tags(self, html):
+        """
+        The capa xml includes many tags such as <additional_answer> or <demandhint> which are not
+        meant to be part of the client html. We strip them all and return the resulting html.
+        """
+        tags = ['demandhint', 'choicehint', 'optionhint', 'stringhint', 'numerichint', 'optionhint',
```

Comment:
It's mostly about extended hints, but not completely. There was one 
case, additional_answer, where the parser layer would remove the node 
during processing, so that it would not show up in the html. I'm making 
a change here that the XML is the XML, and stripping stuff out for the 
get_html path is something done explicitly in one place.

I've added some text at the end of the blurb up at the top about this 
remove-in-one-place change.

You're on the nose about how this could be an easy to fix oversight/bug 
in the future. What's nice is the symptom is soooo obvious. You fire up 
studio or whatever, and this meta stuff shows up in the UI, and then 
there' just the one place to edit to strip that stuff out.

Looking at it with fresh eyes, I'll change the name to 
remove_tags_from_html. My gut is to not make it a constant, figuring 
what makes this maintainable is that it's centralized and well defined, 
but you can have the deciding vote, as I don't feel strongly about it.


On 2/6/15 1:13 PM, Sarina Canelake wrote:
>
> In common/lib/xmodule/xmodule/capa_base.py 
> <https://github.com/edx/edx-platform/pull/6786#discussion_r24272547>:
>
> > @@ -662,6 +688,32 @@ def get_problem_html(self, encapsulate=True):
> >
> >          return htmlout
> > +    def remove_extra_tags(self, html):
> > +        """
> > +        The capa xml includes many tags such as <additional_answer> or <demandhint> which are not
> > +        meant to be part of the client html. We strip them all and return the resulting html.
> > +        """
> > +        tags = ['demandhint', 'choicehint', 'optionhint', 'stringhint', 'numerichint', 'optionhint',
>
> My only reason for wanting it to be a constant is it'll be really easy 
> for someone to overlook adding to this list if they add a new problem 
> type in - but I guess looking at it a second time, these aren't 
> problem types but very specific hint-related capa, so you'd only need 
> to add to this if you were working on hints... right?
>
> If that's the case, I would recommend updating the docstring to 
> indicate how this method is specific to the hinting infrastructure.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/edx/edx-platform/pull/6786/files#r24272547>.
>